boa_engine/src/builtins/temporal/duration/mod.rs

```diff
@@ -0,0 +1,999 @@
+
+/// 7.5.8 `ToTemporalDuration ( item )`
+pub(crate) fn to_temporal_duration(
+    item: &JsValue,
+    context: &mut Context<'_>,
+) -> JsResult<Duration> {
+    // 1a. If Type(item) is Object
+    if item.is_object() {
+        // 1b. and item has an [[InitializedTemporalDuration]] internal slot, then
+        let o = item
+            .as_object()
+            .expect("Value must be an object in this instance.");
+        if o.is_duration() {
+            // a. Return item.
+            let obj = o.borrow();
+            let duration = obj.as_duration().expect("must be a duration.");
+            return Ok(Duration {
+                inner: duration.inner,
+            });
+        }
+    }
+
+    // 2. Let result be ? ToTemporalDurationRecord(item).
+    let result = to_temporal_duration_record(item)?;
+    // 3. Return ! CreateTemporalDuration(result.[[Years]], result.[[Months]], result.[[Weeks]], result.[[Days]], result.[[Hours]], result.[[Minutes]], result.[[Seconds]], result.[[Milliseconds]], result.[[Microseconds]], result.[[Nanoseconds]]).
+    Ok(Duration { inner: result })
+}
+
+/// 7.5.9 `ToTemporalDurationRecord ( temporalDurationLike )`
+pub(crate) fn to_temporal_duration_record(
+    _temporal_duration_like: &JsValue,
+) -> JsResult<DurationRecord> {
+    Err(JsNativeError::range()
+        .with_message("Duration Parsing is not yet complete.")
+        .into())
+}
+
+/// 7.5.14 `CreateTemporalDuration ( years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds [ , newTarget ] )`
+pub(crate) fn create_temporal_duration(
+    record: DurationRecord,
+    new_target: Option<&JsValue>,
+    context: &mut Context<'_>,
+) -> JsResult<JsObject> {
+    // 1. If ! IsValidDuration(years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds) is false, throw a RangeError exception.
+    if !record.is_valid_duration() {
+        return Err(JsNativeError::range()
+            .with_message("Duration values are not valid.")
+            .into());
+    }
+
+    // 2. If newTarget is not present, set newTarget to %Temporal.Duration%.
+    let new_target = if let Some(target) = new_target {
+        target.clone()
+    } else {
+        context
+            .realm()
+            .intrinsics()
+            .constructors()
+            .duration()
+            .constructor()
+            .into()
+    };
+
+    // 3. Let object be ? OrdinaryCreateFromConstructor(newTarget, "%Temporal.Duration.prototype%", « [[InitializedTemporalDuration]], [[Years]], [[Months]], [[Weeks]], [[Days]], [[Hours]], [[Minutes]], [[Seconds]], [[Milliseconds]], [[Microseconds]], [[Nanoseconds]] »).
+    let prototype =
+        get_prototype_from_constructor(&new_target, StandardConstructors::duration, context)?;
+    let mut object = prototype.borrow_mut();
+    let duration = object
+        .as_duration_mut()
+        .expect("prototype must be a Temporal.Duration.");
```

Comment:
This doesn't work because the objectKind is "Ordinary" leading it to fail here.

Reply:
You may still be in the middle of this though and its known this won't work

Reply:
Oh wow. Nice catch! That was just a complete oversight on my part.

Updated it, so the next version should be good to go on this.